src/Runtime/Factors/StringFormatOp.cs

```diff
@@ -681,7 +683,7 @@ public class StringFormatOpBase_NoArgumentNames<TStringFormatOp>
         /// <summary>
         /// Maps the number of argument to an array of default argument names ("0", "1" and so on).
         /// </summary>
-        private static readonly Dictionary<int, string[]> ArgumentCountToNames = new Dictionary<int, string[]>();
+        private static readonly ConcurrentDictionary<int, ReadOnlyArray<string>> ArgumentCountToNames = new ConcurrentDictionary<int, ReadOnlyArray<string>>();
```

Comment:
So, after seing usages... Which all want an interface.
Let's box it to begin with. So: `ConcurrentDictionaty<int, IReadOnlyList<string>>`. Also split this declaration into 2 lines.